### DISCUSSION TITLE
Add CLI option to disable SVG file output

### DISCUSSION THREAD

**Discussion by terrybrash** [terrastruct/d2#779] [Feature Requests]
I use D2 in watch mode and don't want the generated SVG file.

It would be nice to have a CLI option to disable the output entirely, something like `--no-output`.

**Comment by alixander** [comment]
can you describe what you're doing? is it just the browser viewing that you want then?

**Reply by terrybrash** [reply]
Correct. I only view the generated diagrams in the browser, served by the D2 CLI.

**Reply by terrybrash** [reply]
The reason I especially *don't* want output is because I don't want to accidentally check it into source control.

**Reply by alixander** [reply]
hm i can see that use case.

i don't think there needs to be a CLI option though, you can just redirect to /dev/null. if you pass `-` for either the input or output, `d2` reads from stdin or writes to stdout, respectively.

`d2 -w hello.d2 - > /dev/null` should watch without writing anything

**Reply by alixander** [reply]
cc @nhooyr for a second opinion

**Reply by nhooyr** [reply]
I agree that the redirect to /dev/null is sufficient but perhaps we should document it under `--watch`? It's certainly not intuitive as many people don't know about `-`.

Also, maybe worth adding a flag just for windows users where few really know the shell syntax. Like I would have no idea how to redirect to something like /dev/null lol

**Reply by bo-ku-ra** [reply]
> Also, maybe worth adding a flag just for windows users where few really know the shell syntax. Like I would have no idea how to redirect to something like /dev/null lol

linux: /dev/null = windows: nul 

```c:\>d2.exe -w foobar.d2 - >nul```

i think the windows user who does not know the syntax of the shell is not even able to write d2 sources :-p

**Reply by bo-ku-ra** [reply]
> `c:\>d2.exe -w foobar.d2 - >nul`

habit (on windows)
nul: null
con: stdout

> c:\>d2.exe -w foobar.d2 ```con``` >nul

but d2 is not working now. (issues;  dos redirection is not working. #541 )
